interop_binaries/src/bin/janus_interop_aggregator.rs

```diff
@@ -0,0 +1,323 @@
+    _aggregator_id: u8,
+    _hostname_and_port: String,
+}
+
+#[derive(Debug, Serialize)]
+struct EndpointResponse {
+    status: &'static str,
+    endpoint: &'static str,
+}
+
+static ENDPOINT_RESPONSE: EndpointResponse = EndpointResponse {
+    status: "success",
+    endpoint: "/",
+};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct AddTaskRequest {
+    task_id: String,
+    leader: String,
+    helper: String,
+    vdaf: VdafObject,
+    leader_authentication_token: String,
+    #[serde(default)]
+    collector_authentication_token: Option<String>,
+    aggregator_id: u8,
+    verify_key: String,
+    max_batch_lifetime: u64,
+    min_batch_size: u64,
+    min_batch_duration: u64,
+    collector_hpke_config: String,
+}
+
+#[derive(Debug, Serialize)]
+struct AddTaskResponse {
+    status: &'static str,
+    #[serde(default)]
+    error: Option<String>,
+}
+
+async fn handle_add_task(
+    datastore: &Datastore<RealClock>,
+    request: AddTaskRequest,
+) -> anyhow::Result<()> {
+    let task_id_bytes = base64::decode_config(request.task_id, base64::URL_SAFE_NO_PAD)
+        .context("invalid base64url content in \"taskId\"")?;
+    let task_id = TaskId::get_decoded(&task_id_bytes).context("invalid length of TaskId")?;
+    let leader_url = Url::parse(&request.leader).context("bad leader URL")?;
+    let helper_url = Url::parse(&request.helper).context("bad helper URL")?;
+    let vdaf: janus_core::task::VdafInstance = request.vdaf.into();
+    let vdaf: janus_server::task::VdafInstance = vdaf.into();
+    let leader_authentication_token =
+        AuthenticationToken::from(request.leader_authentication_token.into_bytes());
+    let verify_key = base64::decode_config(request.verify_key, URL_SAFE_NO_PAD)
+        .context("invalid base64url content in \"verifyKey\"")?;
+    let min_batch_duration = Duration::from_seconds(request.min_batch_duration);
+    let collector_hpke_config_bytes =
+        base64::decode_config(request.collector_hpke_config, URL_SAFE_NO_PAD)
+            .context("invalid base64url content in \"collectorHpkeConfig\"")?;
+    let collector_hpke_config = HpkeConfig::get_decoded(&collector_hpke_config_bytes)
+        .context("could not parse collector HPKE configuration")?;
+
+    let (role, collector_authentication_tokens) = match (
+        request.aggregator_id,
+        request.collector_authentication_token,
+    ) {
+        (0, None) => {
+            return Err(anyhow::anyhow!("collector authentication is missing"));
+        }
+        (0, Some(collector_authentication_token)) => (
+            Role::Leader,
+            vec![AuthenticationToken::from(
+                collector_authentication_token.into_bytes(),
+            )],
+        ),
+        (1, _) => (Role::Helper, Vec::new()),
+        _ => return Err(anyhow::anyhow!("invalid \"aggregator_id\" value")),
+    };
+
+    let (hpke_config, private_key) = generate_hpke_config_and_private_key(
+        HpkeConfigId::from(0u8),
```

Comment:
Teleporting over from a comment I made on the collector code... This works fine if we assume that HPKE config IDs are unique per-task, but I think that won't be true anymore once https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/289 lands in DAP.